src/Fixer/Whitespace/ReferenceSpacesFixer.php

```diff
@@ -0,0 +1,268 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Whitespace;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\Fixer\ConfigurableFixerInterface;
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolver;
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolverInterface;
+use PhpCsFixer\FixerConfiguration\FixerOptionBuilder;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\FixerDefinitionInterface;
+use PhpCsFixer\Tokenizer\CT;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Jesper Skytte <jesper@skytte.it>
+ */
+final class ReferenceSpacesFixer extends AbstractFixer implements ConfigurableFixerInterface
+{
+    /**
+     * @internal
+     */
+    public const TYPE_ASSIGNMENT = 'assignment';
+
+    /**
+     * @internal
+     */
+    public const TYPE_FUNCTION_SIGNATURE = 'function_signature';
+
+    /**
+     * @internal
+     */
+    public const TYPE_ANONYMOUS_FUNCTION_USE_BLOCK = 'anonymous_function_use_block';
+
+    /**
+     * @internal
+     */
+    public const BY_ASSIGN = 'by_assign';
+
+    /**
+     * @internal
+     */
+    public const BY_REFERENCE = 'by_reference';
+
+    /**
+     * @internal
+     */
+    public const SINGLE_SPACE = 'single_space';
+
+    /**
+     * @internal
+     */
+    public const NO_SPACE = 'no_space';
+
+    /** @var array<'&'|287|288> */
```

Comment:
what `287` stands for?
(hint: numeric value of token ID may very across different PHP versions)